packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -1,202 +1,113 @@
-// @flow
-
-import type {Node} from './types';
-import type {
-  ParcelOptions,
-  Dependency,
-  Target,
-  TransformerRequest
-} from '@parcel/types';
-import type Asset from './Asset';
-import type Config from './Config';
-
+// @flow strict-local
 import EventEmitter from 'events';
-import {
-  AbortController,
-  type AbortSignal
-} from 'abortcontroller-polyfill/dist/cjs-ponyfill';
+
+import type {AssetRequest, ParcelOptions, Target} from '@parcel/types';
 import {PromiseQueue} from '@parcel/utils';
-import OldAssetGraph from './OldAssetGraph';
-import AssetGraph from './AssetGraph';
-import ResolverRunner from './ResolverRunner';
-import WorkerFarm from '@parcel/workers';
 import type {Event} from '@parcel/watcher';
 
-import dumpToGraphViz from './dumpGraphToGraphViz';
+import type Asset from './Asset';
+import AssetGraph from './AssetGraph';
+import type Config from './Config';
+import RequestGraph from './RequestGraph';
+import type {
+  AssetGraphNode,
+  AssetRequestNode,
+  CacheEntry,
+  DepPathRequestNode
+} from './types';
 
-type BuildOpts = {|
-  signal: AbortSignal,
-  shallow?: boolean
-|};
+import dumpToGraphViz from './dumpGraphToGraphViz';
 
 type Opts = {|
   options: ParcelOptions,
   config: Config,
   entries?: Array<string>,
   targets?: Array<Target>,
-  transformerRequest?: TransformerRequest
+  assetRequest?: AssetRequest
 |};
 
 export default class AssetGraphBuilder extends EventEmitter {
-  graph: OldAssetGraph;
+  assetGraph: AssetGraph;
+  requestGraph: RequestGraph;
   queue: PromiseQueue;
-  resolverRunner: ResolverRunner;
   controller: AbortController;
-  farm: WorkerFarm;
-  runTransform: (file: TransformerRequest) => Promise<any>;
   changedAssets: Map<string, Asset>;
 
-  constructor({config, options, entries, targets, transformerRequest}: Opts) {
+  constructor({config, options, entries, targets, assetRequest}: Opts) {
     super();
 
-    this.queue = new PromiseQueue();
-    this.resolverRunner = new ResolverRunner({
+    this.changedAssets = new Map();
+
+    this.assetGraph = new AssetGraph({
+      onNodeAdded: node => this.handleNodeAddedToAssetGraph(node)
+    });
+    this.requestGraph = new RequestGraph({
       config,
-      options
+      options,
+      onAssetRequestComplete: (request, result) =>
+        this.handleCompletedAssetRequest(request, result),
```

Comment:
since they're both single lines, can each of these functions just be inlined directly here?